### DISCUSSION TITLE
Can we please have meaningful commit message in the spirit of open source

### DISCUSSION THREAD

**Discussion by ekagra-ranjan** [NVIDIA/TensorRT-LLM#1448] [General]
Hi TRT LLM team, 

Can we please get meaningful messages in the commit instead of the current generic `Update TensorRT-LLM`? 

It becomes very difficult to track the piece of code responsible for a feature. For e.g., The [release v0.8](https://github.com/NVIDIA/TensorRT-LLM/releases) says `Support for batch manager to return logits from context and/or generation phases` but I cannot find which piece of code in batch manager does this. This prevents the community to find bugs in the code and more importantly track the codebase in order to contribute. Having meaningful commits like any other open source project allows the community to contribute better. 

One way to do this to have smaller PRs with single feature or bug scope. For eg [this PR](https://github.com/NVIDIA/TensorRT-LLM/pull/1358) has 300 files change and is a collection of features but has only 1 commit. Its not possible for people outside Nvidia to find which source code contributes to a particular feature

It would be helpful if we can get meaningful commit messages and release notes pointing to them. Thanks! 

cc: @byshiue @ncomly-nvidia @kaiyux @Shixiaowei02
